ee/katc/structured_clone.go

```diff
@@ -0,0 +1,257 @@
+package katc
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log/slog"
+
+	"golang.org/x/text/encoding/unicode"
+	"golang.org/x/text/transform"
+)
+
+const (
+	tagFloatMax            uint32 = 0xfff00000
+	tagHeader              uint32 = 0xfff10000
+	tagNull                uint32 = 0xffff0000
+	tagUndefined           uint32 = 0xffff0001
+	tagBoolean             uint32 = 0xffff0002
+	tagInt32               uint32 = 0xffff0003
+	tagString              uint32 = 0xffff0004
+	tagDateObject          uint32 = 0xffff0005
+	tagRegexpObject        uint32 = 0xffff0006
+	tagArrayObject         uint32 = 0xffff0007
+	tagObjectObject        uint32 = 0xffff0008
+	tagArrayBufferObjectV2 uint32 = 0xffff0009
+	tagBooleanObject       uint32 = 0xffff000a
+	tagStringObject        uint32 = 0xffff000b
+	tagNumberObject        uint32 = 0xffff000c
+	tagBackReferenceObject uint32 = 0xffff000d
+	tagDoNotUse1           uint32 = 0xffff000e
+	tagDoNotUse2           uint32 = 0xffff000f
+	tagTypedArrayObjectV2  uint32 = 0xffff0010
+	tagMapObject           uint32 = 0xffff0011
+	tagSetObject           uint32 = 0xffff0012
+	tagEndOfKeys           uint32 = 0xffff0013
+)
+
+// structuredCloneDeserialize deserializes a JS object that has been stored in IndexedDB
```

Comment:
Why is it called `structuredCloneDeserialize` ?

Reply:
Named based on the mozilla references below -- could name based on some combination of `firefox` + `indexeddb values` instead

Reply:
I think in the context of indexdb it makes sense to use the mozilla names. But in the context of a DSL for KATC it's much less clear. Something along your handwave there feels better

Reply:
Mostly, the difficulty in naming here for me is that this function and the corresponding one in `ee/indexeddb/values.go` both implement [StructuredDeserialize](https://html.spec.whatwg.org/multipage/structured-data.html#structureddeserialize), which is a great, well-known name. But the functions are not interchangeable -- this one only works for Firefox data, and the `ee/indexeddb` one only works for Chrome data. Maybe we just want something like `structuredDeserializeFirefox` here, since that would be more immediately understandable?

Reply:
Went with `deserializeFirefox`